src/net/java/sip/communicator/impl/protocol/jabber/ChatRoomJabberImpl.java

```diff
@@ -1028,6 +1028,40 @@ public void sendMessage(Message message)
          }
     }
 
+    /**
+     * Sends the <tt>message</tt> with the json-message extension to the
+     * destination indicated by the <tt>to</tt> contact.
+     *
+     * @param message the <tt>Message</tt> to send.
+     * @param json the json message to be sent.
+     * @throws OperationFailedException if sending the message fails for some
+     * reason.
+     */
+    public void sendJsonMessage(Message message, String json)
+            throws OperationFailedException
+    {
+        try
+        {
+            assertConnected();
+            org.jivesoftware.smack.packet.Message msg =
+                    new org.jivesoftware.smack.packet.Message();
+            msg.setType(org.jivesoftware.smack.packet.Message.Type.groupchat);
+            msg.setBody(message.getContent());
+            msg.addExtension(new JsonMessageExtension(json));
+            MessageEventManager.
+                    addNotificationsRequests(msg, true, false, false, true);
+
+            multiUserChat.sendMessage(msg);
+        }
+        catch (NotConnectedException | InterruptedException ex){
+            logger.error("Failed to send JSON message " + message, ex);
```

Comment:
is this log necessary when we throw an OPERATION_FAILED exception?

Reply:
I did not remove it because the sendMessage() method above this uses it in a similar way. I'll remove it now.